File: AEDS/BASES/Ordenacao/quicksort.c
#include <stdio.h>

void swap(int vet[], int i, int j) {
    int aux = vet[i];
    vet[i] = vet[j];
    vet[j] = aux;
}

void quicksort (int vet[], int esq, int dir) {
    int i = esq, j = dir;
    int pivo = vet[(esq + dir) / 2];
    while (i <= j) {
        while (vet[i] < pivo) i++;
        while (vet[j] > pivo) j--;
        if (i <= j) {
            swap(vet, i, j);
            i++; j--;
        }
    }
    if (esq < j) quicksort(vet, esq, j);
    if (i < dir) quicksort(vet, i, dir);
}

File: AEDS/BASES/Ordenacao/quicksort.java
public class quicksort {
    public static void swap (int array[], int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
    public static void quick (int array[], int esq, int dir) {
        int i = esq, j = dir;
        int pivo = array[esq + (dir - esq) / 2];
        while (i <= j) {
            while (array[i] < pivo) i++;
            while (array[j] > pivo) j--;
            if (i <= j) {
                swap(array, i, j);
                i++; j--;
            }
        }
        if (esq < j) quick(array, esq, j);
        if (i < dir) quick(array, i, dir);
    }
}
